Add option to select taunt voice gender

TyroneJohnson opened the pull request:
Allows the player to select the gender for the voice taunts, independent of playermodel.

requested: https://discord.com/channels/614484874199433216/614489592342773790/

TyroneJohnson committed:
Add option to select taunt voice gender

Allows the player to select the gender for the voice taunts, independent of playermodel.

requested: https://discord.com/channels/614484874199433216/614489592342773790/

TyroneJohnson committed:
Move voice gender selection from server to client settings

- Removed server-side voice_gender selection
- Added client-side voice gender selection in player settings
- Implemented dropdown menu for individual voice gender choice
- Enables per-player voice gender customization in multiplayer

Players should be able to choose their voice gender preference.

ZehMatt commented:
I think calling this voice category would be closer to what this is, there are also more options then what you currently have as we pick alyx voice lines for when the selected model is alyx. 

Currently we have:
- alyx
- barney
- monk
- female
- male
- combine
- zombie

So if we go this route we should probably list all the supported choices.

TyroneJohnson committed:
added voice groups and additional taunts

- added individual voice category selection in F4 > Player > Voice menu
- increased number of available taunts
- expand roster of characters with default taunts when voice group is "auto"

TyroneJohnson commented:
Agreed!

ZehMatt committed:
Make sure the choices are ordered correctly, ensure commander entries

ZehMatt committed:
Fix linter warning

ZehMatt committed:
Update changelog.md
